bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/resources/InternalResourceRuleFactory.java

```diff
@@ -0,0 +1,265 @@
+/*******************************************************************************
+ * Copyright (c) 2004, 2022 IBM Corporation and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM - Initial API and implementation
+ *     James Blackburn (Broadcom Corp.) - ongoing development
+ *     Lars Vogel <Lars.Vogel@vogella.com> - Bug 473427
+ *     Christoph Läubrich - Issue #97 - RefreshManager.manageAutoRefresh calls ResourcesPlugin.getWorkspace before the Workspace is fully open
+ *******************************************************************************/
+package org.eclipse.core.internal.resources;
+
+import java.util.HashSet;
+import org.eclipse.core.resources.*;
+import org.eclipse.core.resources.team.FileModificationValidator;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.jobs.ISchedulingRule;
+import org.eclipse.core.runtime.jobs.MultiRule;
+
+/**
+ * Default implementation of IResourceRuleFactory. The teamHook extension
+ * may subclass to provide more specialized scheduling rules for workspace operations that
+ * they participate in.
+ *
+ * @see IResourceRuleFactory
+ * @since 3.0
```

Comment:
It does only add a constructor that takes a workspace, because the original class is public API one can't change it without getting API errors and the SomeClass extends InternalSOmeCLass is used at other places as well in the resource bundle.

Reply:
@iloveeclipse see for example `InternalTeamHook` and `InternalWorkspaceJob` or do should think we should better add a constructor and probably a api filter?

Reply:
Fixed now